csharp/Helpers/EposServiceClient.cs

```diff
@@ -0,0 +1,436 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Net;
+using System.Net.Http.Headers;
+using System.Net.Mime;
+using System.Runtime.Serialization;
+using System.Runtime.Serialization.Json;
+using System.Text;
+using DotNetOpenAuth.Messaging;
+using DotNetOpenAuth.OAuth;
+using DotNetOpenAuth.OAuth.ChannelElements;
+using RD.EposServiceConsumer.Helpers.BusinessObjects;
+
+namespace RD.EposServiceConsumer.Helpers
+{
+    public class EposServiceClient : IEposServiceClient
+    {
+        private static string _key;
+        private static string _secret;
+        private static string _secondSecret;
+        private static string _accessToken;
+        private static TempMemoryTokenManager _tokenManager;
+        private readonly Uri _serviceEndpoint;
+        private static Uri _oAuthEndpoint;
+        private static string _applicationName;
+        private static string _applicationVersion;
+        private static string _testName;
```

Comment:
I think you should remove the `_testName` variable and the overloaded constructor - I'm assuming we've added that to make it easier to track what request was generated by what test, which isn't really relevant to someone trying to use our API.